File: ios/SquatCounterViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(SquatCounterViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onSquat, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onText, RCTDirectEventBlock)
RCT_EXTERN_METHOD(updateFromManager:(nonnull NSNumber *)node)

@end

File: ios/SquatCounterViewManager.swift
import Foundation

@objc(SquatCounterViewManager)
class SquatCounterViewManager: RCTViewManager {
  override func view() -> UIView! {
    return SquatCounterView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func updateFromManager(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! SquatCounterView
      component.update()
    }
  }
}
